### PR TITLE
Embed ROMIO objects into AMPI library

### PR DESCRIPTION
**Pull request by evan-charmworks**

### CODE REVIEW

File: src/libs/ck-libs/ampi/CMakeLists.txt [hunk 1]
```diff
@@ -233,3 +448,51 @@ if(${CMK_AMPI_WITH_ROMIO})
...
+add_library(moduleampi
+    ${ampi-cxx-sources}
+    ${ampi-h-sources}
+    ${romio-objects}
+)
+target_compile_options(moduleampi PRIVATE -DAMPI_NO_UNIMPLEMENTED_WARNINGS -I${CMAKE_BINARY_DIR}/include/ampi/)
+add_dependencies(moduleampi moduletcharm ck)
+
+
+if(${CMK_CAN_LINK_FORTRAN})
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/CMakeLists.txt:462]:
```suggestion
if(CMK_CAN_LINK_FORTRAN)
```

File: src/libs/ck-libs/ampi/CMakeLists.txt [hunk 2]
```diff
@@ -233,3 +448,51 @@ if(${CMK_AMPI_WITH_ROMIO})
...
+        POST_BUILD
+        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/ampif.dir/ampif.C.o ${CMAKE_BINARY_DIR}/lib/ampif.o
+        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/ampifimpl.dir/ampifimpl.f90.o ${CMAKE_BINARY_DIR}/lib/ampifimpl.o
+        COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_CURRENT_BINARY_DIR}/CMakeFiles/ampimod.dir/ampimod.f90.o ${CMAKE_BINARY_DIR}/lib/ampimod.o
+        VERBATIM
+    )
+endif()
+
+
+if(${CMK_AMPI_WITH_ROMIO})
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/CMakeLists.txt:493]:
```suggestion
if(CMK_AMPI_WITH_ROMIO)
```

File: src/libs/ck-libs/ampi/CMakeLists.txt [hunk 3]
```diff
@@ -167,22 +140,265 @@ configure_file(libmoduleampif.dep ${CMAKE_BINARY_DIR}/lib COPYONLY)
...
+    set(romio_dir ${CMAKE_BINARY_DIR}/src/libs/ck-libs/ampi/romio-prefix/src/romio)
+
+    if(CMK_WINDOWS)
+        set(obj_suf ".obj")
+    else()
+        set(obj_suf ".o")
+    endif()
+
+    set(romio-objects
+        ${romio_dir}/adio/ad_nfs/ad_nfs${obj_suf}
```
Comment (matthiasdiener) [inline src/libs/ck-libs/ampi/CMakeLists.txt:155]: Can you check - are the non-{nfs,ufs,testfs} filesystems not built ?

Comment (matthiasdiener) [reply src/libs/ck-libs/ampi/CMakeLists.txt:155]: Nevermind, seems like filesystems other than {nfs,ufs,testfs} need to be explicitly specified during ./configure.

**Review by matthiasdiener — APPROVED** [review]

**Review by stwhite91 — APPROVED** [review]

**Pull request merged**